src/mon/OSDMonitor.cc

```diff
@@ -10848,6 +10848,18 @@ bool OSDMonitor::prepare_command_impl(MonOpRequestRef op,
     err = parse_erasure_code_profile(profile, &profile_map, &ss);
     if (err)
       goto reply;
+    if (profile_map.find("crush-failure-domain") != profile_map.end()) {
+      string failure_domain = profile_map["crush-failure-domain"];
+
+      int type = osdmap.crush->get_type_id(failure_domain);
+      if (type < 0) {
+	ss << "erasure-code-profile " << profile_map
+	  << " contains an invalid failure-domain" << std::endl;
```

Comment:
```suggestion
	ss << "erasure-code-profile " << profile_map
	  << " contains an invalid failure-domain " << std::quote(failure_domain);
```

for 3 reasons:

- no need to add a newline at the end of the output. the client side should take care of the formatting.
- no need to flush the stream
- would be nice to reference the failure domain name which does not exist in crushmap.